tests/nvmeof/test_ceph_nvmeof_neg_tests.py

```diff
@@ -500,365 +608,102 @@ def test_ceph_83576087(ceph_cluster, rbd, pool, config):
         LOG.info("Validation of CEPH-83576087 is successful.")
     except Exception as err:
         raise Exception(err)
-    finally:
-        cleanup_cfg = {
-            "gw_node": config["gw_node"],
-            "disconnect_all": [config["initiator_node"]],
-            "cleanup": ["disconnect_all", "gateway", "pool"],
-            "rbd_pool": pool,
-        }
-        config.update(cleanup_cfg)
-        teardown(ceph_cluster, rbd, nvmegwcli, config)
 
 
-def test_ceph_83576093(ceph_cluster, rbd, pool, config):
+def test_ceph_83576093(ceph_cluster, rbd, nvme_service, pool, config):
     """CEPH-83576093: Perform reboot on GW node and validate the namespaces."""
-    gw_node = get_node_by_id(ceph_cluster, config["gw_node"])
-    ceph = Orch(ceph_cluster, **{})
-    nvmegwcli = create_gateway(
-        nvme_gw_cli_version_adapter(ceph_cluster),
-        gw_node,
-        mtls=config.get("mtls"),
-        shell=getattr(ceph, "shell"),
-        port=config.get("gw_port", 5500),
-        gw_group=config.get("gw_group"),
-    )
-
-    subsystem = dict()
-    listener_port = find_free_port(gw_node)
-    subsystem.update(
-        {
-            "nqn": "nqn.2016-06.io.spdk:ceph-83576093",
-            "serial": 113,
-            "listener_port": listener_port,
-            "allow_host": "*",
-        }
-    )
-    initiator_cfg = {
-        "subnqn": "nqn.2016-06.io.spdk:ceph-83576093",
-        "listener_port": listener_port,
-        "node": config["initiator_node"],
-    }
+    client = get_node_by_id(ceph_cluster, config["initiator_node"])
+    initiator = NVMeInitiator(client)
 
+    gw_node = get_node_by_id(ceph_cluster, config["gw_node"])
     try:
-        configure_subsystems(ceph_cluster, rbd, pool, nvmegwcli, subsystem)
-        name = generate_unique_id(length=4)
+        # Deploy NVMe Service
+        if config.get("install"):
+            nvme_service.deploy()
 
-        # Create image
-        img = f"{name}-image"
-        rbd.create_image(pool, img, "1G")
-        ns_args = {
-            "args": {"subsystem": subsystem["nqn"], "rbd-pool": pool, "rbd-image": img}
-        }
-        nvmegwcli.namespace.add(**ns_args)
-
-        config.update(initiator_cfg)
-        client = get_node_by_id(ceph_cluster, config["node"])
-        initiator = NVMeInitiator(client)
+        # Initialize gateways
+        nvme_service.init_gateways()
+        nvmegwcli = nvme_service.gateways[0]
         cmd_args = {
             "transport": "tcp",
             "traddr": nvmegwcli.node.ip_address,
-            "trsvcid": listener_port,
+            "trsvcid": config["initiators"][0]["listener_port"],
         }
 
-        json_format = {"output-format": "json"}
-        _dir = f"/tmp/dir_{generate_unique_id(4)}"
-        _file = f"{_dir}/test.log"
+        # Configure Subsystems
+        configure_subsystems(nvme_service)
```

Comment:
Can we not directly call configure_gw_entities and that will take care of configuring subsystems, listeners and hosts right? All you need to do is pass those keywords to config I guess. Can we check this?